pytorch_lightning/strategies/ddp_spawn.py

```diff
@@ -186,6 +189,10 @@ def configure_ddp(self) -> None:
         self.model = self._setup_model(LightningDistributedModule(self.model))
         self._register_ddp_hooks()
 
+        # set up optimizers after the wrapped module has been moved to the device
+        self.setup_optimizers(self.lightning_module.trainer)
+        optimizers_to_device(self.optimizers, self.root_device)
```

Comment:
why not place it in the setup method as with the DDPStrategy? These two will need to collapse anyway in the future.